docs-src/prod-readiness-context/data-encryption.md

```diff
@@ -0,0 +1,54 @@
+
+- Inputs and outputs/results in your [Workflow](/concepts/what-is-a-workflow-execution), [Activity](/concepts/what-is-an-activity-execution), and [Child Workflow](/concepts/what-is-a-child-workflow-execution)
+- [Signal](/concepts/what-is-a-signal) inputs
+- [Memo](/concepts/what-is-a-memo)
+- Headers (verify if applicable to your SDK)
+- [Query](/concepts/what-is-a-query) results
+- Results of [Local Activity](/concepts/what-is-a-local-activity), [Side Effects](/concepts/what-is-a-side-effect)
+- [Application errors and failures](/kb/failures)
+
+Using encryption ensures that your sensitive data exists unencrypted only on the Client and the Worker Process that is executing the Workflows and Activities, on hosts that you control.
+
+By default, your data is serialized to a [Payload](/concepts/what-is-a-payload) by a [Data Converter](/concepts/what-is-a-data-converter).
+To encrypt your Payload, configure your custom encryption logic with a [Payload Codec](/concepts/what-is-a-payload-codec) and set it with a [custom Data Converter](/concepts/what-is-a-custom-data-converter).
+
+A Payload Codec does byte-to-byte conversion to transform your Payload (for example, by implementing compression and/or encryption and decryption) and is an optional step that happens between the wire and the [Payload Converter](/concepts/what-is-a-payload-converter):
+
+```bash
+User code <--> Payload Converter <--> Payload Codec <--> Wire <--> Temporal Server
+```
+
+You can run your Payload Codec with a [Codec Server](/concepts/what-is-a-codec-server) and use the Codec Server endpoints in Web UI and tctl to decode your encrypted Payload locally.
+
+However, if you plan to set up [remote data encoding](/concepts/what-is-remote-data-encoding) for your data, ensure that you consider all security implications of running encryption remotely before implementing it.
+
+In codec implementations, we recommend running the function (such as compression or encryption) on the entire input Payload and putting the result in the data field of a new Payload with a different encoding metadata field.
+Using this technique ensures that the input Payload's metadata is preserved.
+When the encoded Payload is sent to be decoded, you can verify the metadata field before applying the decryption.
+If your Payload is _not_ encoded, we recommend passing the unencoded data to the decode function instead of failing the conversion.
+
+Examples for implementing encryption:
+
+- [Go sample](https://github.com/temporalio/samples-go/tree/main/encryption)
+- [Java sample](https://github.com/temporalio/samples-java/tree/main/src/main/java/io/temporal/samples/encryptedpayloads)
+- [Python sample](https://github.com/temporalio/samples-python/tree/main/encryption)
+- [TypeScript sample](https://github.com/temporalio/samples-typescript/tree/main/encryption)
+
+References for implementing compression:
```

Comment:
```suggestion
Examples for implementing compression (note that better compression algorithms should probably be used):
```

Reply:
lol, I'll reframe it to "Examples for implementing compression" but leave the note part out. Best to let the reader decide if there are better compression algorithms they can use; this is just an example set for reference.